front/pages/api/w/[wId]/invitations/[iId]/index.ts

```diff
@@ -88,10 +92,30 @@ async function handler(
       }
       const body = bodyValidation.right;
 
-      invitation = await updateInvitationStatus(auth, {
-        invitation,
-        status: body.status,
-      });
+      if (body.status) {
+        invitation = await updateInvitationStatus(auth, {
+          invitation: invitation,
+          status: body.status,
+        });
+      }
+      if (body.initialRole) {
+        const retrievedInvitation = await MembershipInvitation.findOne({
```

Comment:
Let's not use models in API routes as much as possible.

Instead create an updateInvitationStatus or change it into an updateInvationStatusAndRole